simulator/event_manager.go

```diff
@@ -179,7 +185,7 @@ type EventHistoryCollector struct {
 
 	m    *EventManager
 	page *ring.Ring
```

Comment:
Checked all reads/write to `page` and it's mostly protected but not in `GetLatestPage()`. Not sure if there could be a race between `GetLatestPage()` and e.g. `PostEvents()` which does acquire a lock to protect the `page`:

```go
for _, c := range m.collectors {
		ctx.WithLock(c, func() {
			if c.eventMatches(req.EventToPost) {
				c.page = c.page.Prev()
				c.page.Value = req.EventToPost
				Map.Update(c, []types.PropertyChange{{Name: "latestPage", Val: c.GetLatestPage()}})
			}
		})
	}
```

Reply:
`c.GetLatestPage()` reads from `c.page`, but is also within the `ctx.WithLock(c,...)`.

And API access to `GetLatestPage()` is via the PropertyCollector, which locks the object here: https://github.com/vmware/govmomi/blob/fe8ef53a7a43c875e08bfb1c119b698073fe0376/simulator/property_collector.go#L441

Reply:
Thx, concurrent access to GetLatestPage via API was my concern, so looks like we're good 👍